<launch>

<arg name="ip_addr" default="192.168.5.100"/>
<arg name="user_name" default="admin"/>
<arg name="password" default="ht123456"/>
<arg name="port" default="8000"/>
<arg name="channel" default="1"/>
<!--link_mode: 0-tcp, 1-udp, 2-multicast, 3-rtp, 4-rtp/rtsp, 5-rstp/http-->
<arg name="link_mode" default="0"/>


<arg name="image_width" default="1280"/>
<arg name="image_height" default="720"/>
<arg name="camera_frame_id" default="hik_camera"/>
<arg name="camera_name" default="hik_camera"/>
<arg name="camera_info_url" default=""/>

<node pkg="hikvision_ros" type="hik_cam_node" name="hik_cam_node" output="screen">

    <!-- modify LD_LIBRARY_PATH for loading hikvision sdk -->
    <env name="LD_LIBRARY_PATH"
         value="$(env LD_LIBRARY_PATH):$(find hikvision_ros)/lib:$(find hikvision_ros)/lib/HCNetSDKCom"/>

    <param name="ip_addr" value="$(arg ip_addr)" />
    <param name="user_name" value="$(arg user_name)" />
    <param name="password" value="$(arg password)" />
    <param name="port" value="$(arg port)" />
    <param name="channel" value="$(arg channel)"/>
    <param name="link_mode" value="$(arg link_mode)"/>

    <param name="image_width" value="$(arg image_width)"/>
    <param name="image_height" value="$(arg image_height)"/>
    <param name="camera_frame_id" value="$(arg camera_frame_id)"/>
    <param name="camera_name" value="$(arg camera_name)"/>
    <param name="camera_info_url" value="$(arg camera_info_url)"/>
</node>


</launch>
